server/src/router.ts

```diff
@@ -0,0 +1,156 @@
+                const {hexalotId} = req.params
+                res.locals.hexalotId = hexalotId
+                next()
+            },
+            hexalotRoute,
+        )
+
+    islandRoute
+        .get("/", async (req, res) => {
+            res.json(
+                await store.getPattern(res.locals.island.islandName),
+            )
+        })
+        .post(
+            "/claim-lot",
+            [
+                hexalotIDValidation(body("id")),
+                body("x").isInt().toInt(),
+                body("y").isInt().toInt(),
+                body("genomeData").isJSON(),
+                validateRequest,
+            ],
+            async (req: Request, res: Response) => {
+                const {
+                    x,
+                    y,
+                    genome,
+                    id,
+                } = req.body
+                const island: Island = res.locals.island
+                await island.load()
+
+                // TODO: probably gonna wanna mutex this
+                try {
+                    await island.claimHexalot(
+                        {x, y},
+                        id,
+                        genome,
+                    )
+                } catch (err) {
+                    res.status(HttpStatus.BAD_REQUEST).json({errors: [err.toString()]})
+                    return
+                }
+                res.json(island.pattern)
+            },
+        )
+
+    hexalotRoute
+        .route("/genome-data")
+        .get(async (req, res) => {
+            const genomeData = await store.getGenome(res.locals.hexalotId)
+            if (!genomeData) {
+                res.sendStatus(HttpStatus.NOT_FOUND)
+                return
+            }
+            res.json(genomeData)
+        })
+        .post(
+            [
+                body("genomeData").isJSON(),
+                validateRequest,
+            ],
+            async (req: Request, res: Response) => {
+                await store.setGenome(res.locals.hexalotId, req.body.genomeData)
+                res.sendStatus(HttpStatus.OK)
+            },
+        )
+
+    hexalotRoute
+        .route("/rotation")
+        .get(async (req, res) => {
+            const rotation = await store.getRotation(res.locals.hexalotId)
+            res.json(rotation)
+        })
+        .post(
+            [
+                body("rotation").isNumeric().toFloat(),
+            ],
+            async (req: Request, res: Response) => {
+                await store.setRotation(res.locals.hexalotId, req.body.rotation)
+                res.sendStatus(HttpStatus.OK)
+            },
+        )
+
+    hexalotRoute
+        .route("/journey")
+        .get(async (req, res) => {
+            const rotation = await store.getRotation(res.locals.hexalotId)
+            res.json(rotation)
+        })
+        .post(
+            [
+                body("rotation").isNumeric().toFloat(),
+            ],
+            async (req: Request, res: Response) => {
```

Comment:
hmmm.. gonna have to look this up.  stuff i haven't used.

Reply:
You mean middlewares & whatnot? Pretty cool

Reply:
async and await

Reply:
it's so much nicer than callback or Promise hell